src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -218,6 +220,21 @@ public Condition newCondition() {
     @Deprecated
     public abstract void delete(DeleteByQuery delete) throws EngineException;
 
+    /**
+     * Attempts to do a special commit where the given syncID is put into the commit data. The attempt
+     * succeeds if there are not pending writes in lucene and the current point is equal to the expected one.
+     * @param syncId id of this sync
+     * @param expectedCommitId the expected value of
+     * @return true if the sync commit was made, false o.w.
+     */
+    public abstract SyncedFlushResult syncFlushIfNoPendingChanges(String syncId, byte[] expectedCommitId) throws EngineException;
```

Comment:
I don't like the fact that we have an impl detail in the method name maybe we have a way to syncFlush if there are pending changes? just call it `syncFlush`?